libcaf_io/src/io/basp_broker.cpp

```diff
@@ -329,10 +334,46 @@ behavior basp_broker::make_behavior() {
       }
       return std::make_tuple(x, std::move(addr), port);
     },
-    [=](tick_atom, size_t interval) {
+    [=](tick_atom, actor_clock::time_point scheduled,
+        timespan heartbeat_interval, timespan connection_timeout) {
+      auto now = clock().now();
+      if (now < scheduled) {
+        CAF_LOG_WARNING("received tick before its time, reschedule");
+        scheduled_send(this, scheduled, tick_atom::value, scheduled,
+                       heartbeat_interval, connection_timeout);
+        return;
+      }
+      auto next_tick = scheduled + heartbeat_interval;
+      if (now >= next_tick) {
+        CAF_LOG_ERROR("Lagging a full heartbeat interval behind! "
+                      "Interval too low or BASP actor overloaded! "
+                      "Other nodes may disconnect.");
+        while (now >= next_tick)
+          next_tick += heartbeat_interval;
+
+      } else if (now >= scheduled + (heartbeat_interval / 2)) {
+        CAF_LOG_WARNING("Lagging more than 50% of a heartbeat interval behind! "
+                        "Interval too low or BASP actor overloaded!");
+      }
+      // Send out heartbeats.
       instance.handle_heartbeat(context());
```

Comment:
I think `handle_heartbeat` is a misleading name. The function only sends out heartbeat messages. How about renaming it to `send_heartbeats` or `write_heartbeats`?